### PR TITLE
Actually delay running of mount start jobs when /p/s/mountinfo is rate limited

### PR DESCRIPTION
**Pull request by msekletar**

### CODE REVIEW

File: src/core/unit.c [hunk 1]
```diff
@@ -1868,6 +1868,10 @@ int unit_start(Unit *u) {
 
         assert(u);
 
+        /* Let's hold off running start jobs for mount units when /proc/self/mountinfo monitor is rate limited. */
+        if (u->type == UNIT_MOUNT && sd_event_source_is_ratelimited(u->manager->mount_event_source))
+                return -EAGAIN;
```
Comment (yuwata) [inline src/core/unit.c:1873]: Maybe, it is better to introduce a function in unit vtable, e.g. `.can_start_early()`??

**Review by yuwata — APPROVED** [review]
LGTM. One minor suggestion.

File: src/core/unit.c (hunk 1 above)
Comment (msekletar) [reply src/core/unit.c:1873]: I don't know if it is worth adding virtual function if we now that only mount unit type would provide an implementation.

Comment (yuwata) [reply src/core/unit.c:1873]: OK. Then let's merge this as is, and introduce the vtable entry later when necessary.

**Pull request merged**